packages/protvista-structure/src/position-mapping.ts

```diff
@@ -41,51 +43,73 @@ const translatePositions = (
   end: number;
   entity: number;
   chain: string;
-} | null => {
+}[] => {
   // Return if mappings not ready
   if (!mappings) {
-    return null;
+    return [];
   }
   // return if they have been set to 'undefined'
   if (!start || !end || Number.isNaN(start) || Number.isNaN(end)) {
     throw new PositionMappingError("Invalid start, end coordinates");
   }
-  /* eslint-disable no-restricted-syntax */
-  for (const mapping of mappings) {
-    if (
-      mapping.unp_end - mapping.unp_start !==
-      mapping.end.residue_number - mapping.start.residue_number
-    ) {
-      throw new PositionMappingError(
-        "Mismatch between protein sequence and structure residues"
-      );
-    }
-    if (
-      (mappingDirection === "UP_PDB" &&
-        (start < mapping.unp_start || end > mapping.unp_end)) ||
-      (mappingDirection === "PDB_UP" &&
-        (start < mapping.start.residue_number ||
-          end > mapping.end.residue_number))
-    ) {
-      throw new PositionMappingError(
-        "Start or end coordinate outside of mapping range"
-      );
-    }
-    // TODO: this is wrong because there are gaps in the PDB sequence though if
-    // this is the case PositionMappingError would have already been thrown and
-    // we should not reach this point.
-    const offset =
-      mappingDirection === "UP_PDB"
-        ? mapping.start.residue_number - mapping.unp_start
-        : mapping.unp_start - mapping.start.residue_number;
-    return {
-      entity: mapping.entity_id,
-      chain: mapping.chain_id,
-      start: start + offset,
-      end: end + offset,
-    };
+  // return a translation separately for each chain (if it exists)
+  const translations = Object.entries(
```

Comment:
Group the list of mapping regions by their chain, then iterate over the chain mapping regions. If a translation for the given chain is found, it will be added to the list. If it's not found, we return null which will be filtered out. If a mapping is not found for any chain, we throw an exception, same as before.